app/components/PrivateKeyView.jsx

```diff
@@ -16,21 +16,49 @@ export default class PrivateKeyView extends Component {
     constructor() {
         super();
         this.state = this._getInitialState();
+
+        this.showModal = this.showModal.bind(this);
+        this.hideModal = this.hideModal.bind(this);
+
+        this.showQrModal = this.showQrModal.bind(this);
+        this.hideQrModal = this.hideQrModal.bind(this);
+
+        this.onClose = this.onClose.bind(this);
     }
 
     _getInitialState() {
-        return {wif: null};
+        return {
+            isModalVisible: false,
+            isQrModalVisible: false,
```

Comment:
The more I read this PR the more I am wondering if the Modal can't handle the visibility by itself without parent state, and only do on button click callbacks where necessary.

Was this evaluated? If so, was there anything against such an approach?